### PR TITLE
Move checkObjectCoercible out of runtime.js

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/runtime/checkObjectCoercible.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const $TypeError = TypeError;
+
+// https://people.mozilla.org/~jorendorff/es6-draft.html#sec-checkobjectcoercible
+export default function checkObjectCoercible(argument) {
+  if (argument == null) {
```
Comment (johnjbarton) [inline src/runtime/checkObjectCoercible.js]: `===`

**Comment by johnjbarton** [comment]
LGTM

File: src/runtime/checkObjectCoercible.js (hunk 1 above)
Comment (arv) [inline src/runtime/checkObjectCoercible.js]: That needs to be `==` but I can change to compare to both null and undefined if you care?

Comment (johnjbarton) [inline src/runtime/checkObjectCoercible.js]: or just a comment so we remember that conversion was intended.

**Pull request merged**